pkg/flare/archive.go

```diff
@@ -157,6 +158,11 @@ func createArchive(fb flarehelpers.FlareBuilder, confSearchPaths SearchPaths, lo
 
 	fb.RegisterFilePerm(security.GetAuthTokenFilepath())
 
+	if runtime.GOOS == "linux" {
+		fb.RegisterDirPerm(filepath.Dir(config.Datadog.GetString("system_probe_config.sysprobe_socket")))
+	}
+	fb.RegisterDirPerm(config.Datadog.GetString("system_probe_config.bpf_dir"))
+
```

Comment:
If those 2 are linux specific we should move them to archive_nix.go or maybe a system_probe dedicated file instead of checking for GOOS.
We want to group OS or product specific part together.

Reply:
only `system_probe_config.sysprobe_socket` is linux specific

Reply:
added a tweak, let me know if I can improve that